src/mqt/predictor/rl/predictorenv.py

```
@@ -289,7 +301,87 @@ def export_circuit(self, qc: QuantumCircuit | None = None) -> QuantumCircuit:
         return exported
 
-    def _log_step_reward(self, step_index: int, action_name: str, reward_val: float, done: bool) -> None:
+    def _collect_tracer_data(
+        self,
+        step_index: int,
+        action_name: str,
+        reward_val: float,
+        fom_value: float,
+        fom_kind: str,
+        feature_vector: dict[str, int | NDArray[np.float32]],
+        done: bool,
```

Ruff (0.15.10)

[warning] Boolean-typed positional argument in function definition

(FBT001)